app/controllers/questions_controller.rb

```diff
@@ -57,13 +65,33 @@ def destroy
   end
 
   private
+
+  def set_test
+    @test = Test.find(params[:test_id])
+  end
     # Use callbacks to share common setup or constraints between actions.
     def set_question
       @question = Question.find(params[:id])
     end
 
     # Only allow a list of trusted parameters through.
     def question_params
-      params.require(:question).permit(:question, :answer, :test_id)
+      params.require(:question).permit(:question, :question_id)
+    end
+
+    def initialize_search
+      @question = Question.order(question: :desc)
+      session[:question] ||= params[:question]
+      session[:filter] = params[:filter]
+      params[:filter_option] = nil if params[:filter_option] == ""
+      session[:filter_option] = params[:filter_option]
+    end
```

Comment:
You have pretty same method across the controllers. Consider moving it somewhere to keep it DRY.